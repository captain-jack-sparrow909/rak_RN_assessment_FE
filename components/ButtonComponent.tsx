import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

interface ButtonComponentProps {
    title: string;
    onPress: () => void;
    style?: object;
    textStyle?: object;
    logoName?: string;
    }

const ButtonComponent = ({ title, onPress, style, textStyle, logoName }: ButtonComponentProps) => {
  return (
    <TouchableOpacity style={[styles.button, style]} onPress={onPress}>
      {logoName && <Text><Ionicons name={logoName as any} size={20}/></Text>}
      <Text style={[styles.buttonText, textStyle]}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#E63946',
    paddingVertical: 15,
    borderRadius: 8,
    alignItems: 'center',
    marginVertical: 10,
    minWidth: 380,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    gap: 5,
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
});

export default ButtonComponent;
