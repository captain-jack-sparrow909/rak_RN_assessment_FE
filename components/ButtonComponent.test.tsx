import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ButtonComponent from './ButtonComponent';

describe('ButtonComponent', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<ButtonComponent title="Test Button" onPress={() => {}} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call onPress when button is pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(<ButtonComponent title="Test Button" onPress={onPressMock} />);
    fireEvent.press(getByText('Test Button'));
    expect(onPressMock).toHaveBeenCalled();
  });

});