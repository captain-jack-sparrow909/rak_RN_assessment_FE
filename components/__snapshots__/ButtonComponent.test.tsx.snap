// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#E63946",
      "borderRadius": 8,
      "display": "flex",
      "flexDirection": "row",
      "gap": 5,
      "justifyContent": "center",
      "marginVertical": 10,
      "minWidth": 380,
      "opacity": 1,
      "paddingVertical": 15,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Test Button
  </Text>
</View>
`;
